File: templates/go/starter.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

/*
Starter template for Go (latest stable)
Entrypoint: Run with `go run starter.go`
Build with: `go build starter.go`
*/

func main() {
	/*
		Sample I/O:
		  Input: (via stdin or args)
		  Output: (via stdout)
	*/

	// Read input
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter your name: ")
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)

	// Process
	result := fmt.Sprintf("Hello, %s!", input)

	// Write output
	fmt.Println(result)
}

File: templates/csharp14/Starter.cs
using System;

/*
Starter template for C# 14 (.NET 10.0)
Entrypoint: Run with `dotnet run` (requires a .csproj) or compile with `csc Starter.cs` and run `./Starter`
For simpler testing: `dotnet script Starter.cs` (requires dotnet-script tool)
Sample I/O:
  Input: (via stdin or args)
  Output: (via stdout)
*/

class Starter
{
    static void Main(string[] args)
    {
        // Read input
        Console.Write("Enter your name: ");
        string? input = Console.ReadLine();

        // Process
        string result = $"Hello, {input ?? "World"}!";

        // Write output
        Console.WriteLine(result);
    }
}
